docs/architecture/joins.txt

```diff
@@ -119,13 +127,13 @@ loop to acquire the data before being able to join. After finding the locations
 of the required shards (which is done in the planning stage), the smaller data
 set (based on the row count) is broadcast amongst all the nodes holding the
 shards they are joined with. After that, each of the receiving nodes can start
-running a nested loop on the subset it has just received. Finally, the results
-are pushed to the original (planner) node to merge and return the results to
-the requesting client (see
-:ref:`joins_figure_2`).
+running a nested loop on the subset it has just received. Finally, these
+intermediate results are pushed to the original (planner) node to merge and
+return the results to the requesting client (see :ref:`joins_figure_2`).
 
-If the rows in the join result from or in another (nested) join or use a `table
-function`_ , the data is broadcast from (or to) different nodes directly.
+If the source rows of a join originate from another join result (joining more
+than 2 tables) or use a `table function`_ , the data is broadcast from (or to)
+different nodes directly.
```

Comment:
I think this is still a bit confusing. It's not really clear who the "different" nodes are

Reply:
I'm not sure if this is the correct meaning but how about:
```
the data is broadcast from the source to target nodes of directly.
```

Reply:
Yeah, I think this is better. Maybe we should specify who the source/target is?
Eg. 
`the data is broadcasted from the source join nodes directly to the nodes executing the current join.`

Reply:
I'm thinking about removing this completely as it may confuse more than it helps.